packages/ra-core/src/dataProvider/useDataProvider.spec.tsx

```diff
@@ -251,6 +259,63 @@ describe('useDataProvider', () => {
         expect(queryByTestId('error')?.textContent).toBe(
             'ra.notification.data_provider_error'
         );
+
+        process.env = originalEnv;
+    });
+
+    it('should call getList and not show error in test environment', async () => {
+        jest.resetModules();
+        process.env = {
+            ...originalEnv,
+        };
```

Comment:
You can probably put all the new tests inside a `describe` with a `beforeEach` and `afterEach` to ensure nothing leaks.